### PR TITLE
Fixes Image selection dropdown scroll issue

### PR DESCRIPTION
**Pull request by manaswinidas**
Closes: https://issues.redhat.com/browse/RHOAIENG-3412

## Description

Fixes the image selection dropdown scroll issue. Upgraded `SimpleDropdownSelect` component to PF5.

Screenshot 2024-04-10 at 7 23 06 PM

https://github.com/opendatahub-io/odh-dashboard/assets/22885912/3d8af954-c3cf-4d6c-8702-b14b5096b359

## How Has This Been Tested?

Try creating a workbench after creating many custom images using the `Notebook images` section in the `Settings` section in the Dashboard. You should see a scroll bar while trying to select images in the `Image selection`, which doesn't break the layout. 

## Test Impact

Fixed failing tests due to the PF5 migration. Added new tests for scrollable dropdown selection

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [X] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by lucferbux — CHANGES_REQUESTED** [review]
Ok a couple of things here:

Have you tested wether this fix the issue? I would think it's still happening if we still have the css.

We should have coverage with mockup data that might be recreating the issue, we can talk about that async if you want.

Is this refactor gonna help other instances of this? Maybe we should just fix the issue of the height and do the refactor in another issue OR just do the refactor in all the instances, but it's not a great idea having both components.

**Review by dpanshug — APPROVED** [review]
/lgtm

**Comment by manaswinidas** [comment]
@lucferbux Fixed all test failures arising due to the migration. Added a new test to test scrollable dropdown selection as requested. Even manually tested the views on the pages, seems to work fine. The CSS file you mentioned seems to fix some width issues instead of height and I think it's required coz the extra padding persists if we remove that CSS. This is ready for review.

File: frontend/src/__tests__/cypress/cypress/support/commands/application.ts [hunk 1]
```diff
@@ -27,6 +26,12 @@ declare global {
        */
       findDropdownItem(name: string | RegExp): Cypress.Chainable<JQuery>;
 
+      /**
+       * Finds a patternfly dropdown item by data-testid, first opening the dropdown if not already opened.
+       *
+       * @param testId the name of the item
+       */
+      findByDropdownItemId(testId: string): Cypress.Chainable<JQuery>;
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/support/commands/application.ts:34]:
```suggestion
      findDropdownItemByTestId(testId: string): Cypress.Chainable<JQuery>;
```

**Review by lucferbux — APPROVED** [review]
/lgtm

Had to check the part of `testId` prop, but I think that's the only solution we can go for now, since that's how patternfly recommends using the testing id.

**Pull request merged**